fuzzing/libfuzzer/libs/utils.py

```diff
@@ -0,0 +1,228 @@
+import sys
+import yaml
+import time
+import shutil
+import socket
+import psutil
+import subprocess
+import lsb_release
+import csv
+from datetime import datetime as dt
+import collections
+import pkg_resources
+import socket
+import netifaces as ni
+import re
+from common.config.constants import *
+from threading import Timer
+import shlex
+
+
+def exec_shell_cmd(cmd, stdout_val=subprocess.PIPE):
+    try:
+        cmd_stdout = subprocess.run([cmd], shell=True, check=True, stdout=stdout_val, stderr=subprocess.STDOUT, universal_newlines=True)
+
+        if stdout_val is not None and cmd_stdout.stdout is not None:
+            return cmd_stdout.stdout.strip()
+
+        return cmd_stdout
+
+    except subprocess.CalledProcessError as e:
+        print(e.output)
+
+
+def write_log(content, log_file):
+    log = f"{LIBFUZZER_LOGS_DIR}/{log_file}.log"
+    fd = open(log, "a+")
+    fd.write(content)
+    fd.write("\n")
+    fd.close()
+    print("logs contents are copied " + log)
+
+
+def exec_shell_popen(cmd, log_file, timeout=0):
+    print(f"Process started by running this command : {cmd}")
+    process = subprocess.Popen(shlex.split(cmd), stdout=subprocess.PIPE, encoding="utf-8")
+    process_output = ''
+    try:
+        if timeout!=0:
+            timer = Timer(timeout, process.kill)
+            timer.start()
+        while True:
+            output = process.stdout.readline()
+            print(output)
+            process_output += output
+            if process.poll() is not None and output == '':
+                break
+    finally:
+        if timeout!=0:
+            timer.cancel()
+        process.stdout.close()
+        write_log(process_output, log_file)
+    return process.returncode
+
+
+def read_config_yaml(config_file_path):
+    with open(config_file_path, "r") as config_fd:
+        try:
+            config_dict = yaml.safe_load(config_fd)
+        except yaml.YAMLError as exc:
+            raise Exception(exc)
+    return config_dict
+
+
+def get_distro_and_version():
+    distro = lsb_release.get_distro_information().get('ID').lower()
+    distro_version = lsb_release.get_distro_information().get('RELEASE')
+    return distro, distro_version
+
+
+def clear_system_cache():
```

Comment:
this is for perf, for libfuzzer you don't need this.

Reply:
removed